geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -65,6 +67,35 @@ public RedisSet(Collection<byte[]> members) {
    */
   public RedisSet() {}
 
+  public RedisSet(int size) {
+    this.members = new RedisSet.MemberSet(size);
+  }
+
+  public Set<byte[]> sdiff(RegionProvider regionProvider, List<RedisKey> keys) {
```

Comment:
I don't like sdiff being an instance method on RedisSet. It seems like the implementation counts on it always starting with an empty RedisSet. And that is how we use it. So it seems like the only RedisSet instances we care about are the existing ones we will read. So I think it is a bit confusing to create an empty RedisSet instance and then throw it away after we get the members out of it. I think it would be better for sdiff and calculateDIff to be "static" and the "members" in calculateDiff to be a local variable that is created like so: "Set<byte[]> members = new MemberSet(firstSet.members);". This will also give you a better initial size that will do fewer rehashes.
Also you should probably rename "members" to something like "diff".
Instead of static methods on RedisSet these could be on SetOpExecutor but since they read the internals of the RedisSet instances I think it is better on RedisSet.